tools/memory_monitor.py

```diff
@@ -348,14 +354,24 @@ def _subtract_first_element(data):
     parser.add_argument("executable", type=str, nargs="+", help="Target executable to monitor memory for.")
     args = parser.parse_args()
 
-    def log(mm, fz):
-        mm.save_memory_logs(
-            *mm.get_data(memory_from_zero=fz), save_dir=Path(args.log_dir), filename_suffix="_from-zero" if fz else ""
-        )
-
-    for memory_type, mem_from_zero in [(MemoryType.RSS, False), (MemoryType.SYSTEM, False), (MemoryType.SYSTEM, True)]:
-        memory_monitor = MemoryMonitor(memory_type=memory_type, include_child_processes=True)
-        memory_monitor.start(at_exit_fn=partial(log, memory_monitor, mem_from_zero))
+    memory_monitors = [
+        MemoryMonitor(memory_type=mt, include_child_processes=True).start()
+        for mt in (MemoryType.RSS, MemoryType.SYSTEM)
+    ]
 
     with subprocess.Popen(" ".join(args.executable), shell=True) as p:
         p.wait()
+
+        # Stop addition of new values as soon as possible
+        for mm in memory_monitors:
+            mm._monitoring_thread_should_stop = True
+
+    for mm in memory_monitors:
+        mm.stop()
+        for fz in (True, False):
+            time_values, memory_values = mm.get_data(memory_from_zero=fz)
+            # Most probably the last value is recorded once the child process has already died
+            time_values, memory_values = time_values[:-1], memory_values[:-1]
+            mm.save_memory_logs(
```

Comment:
What you think about add summary to stdout, somthing like like:

```python
from tabulate import tabulate
...
    summary_data = []
    for mm in memory_monitors:
    ...
            summary_data.append([mm.memory_type.value, fz, max(memory_values)])
    print(tabulate(summary_data, headers=["Memory type", "From zero", "Pick"]))
```
```
Memory type    From zero          Pick
-------------  -----------  ----------
rss            True            41.7188
rss            False          103.848
system         True            41.6367
system         False        71775.9
```

Reply:
Added